src/renderer/src/hooks/useGetFullTransactions.tsx

```diff
@@ -0,0 +1,333 @@
+    initialPageParam: 1,
+    getNextPageParam: ({ page }) => page,
+  })
+
+  const data = useMemo(() => {
+    if (query.isLoading) return []
+
+    let groupedData = new Map<string, TFullTransactionsItem>()
+    const data = cloneDeep(query.data?.pages?.flatMap(page => page?.data ?? []) ?? [])
+
+    data.forEach(items => {
+      groupedData = new Map<string, TFullTransactionsItem>([...groupedData, ...items])
+    })
+
+    const insertedPendingTransferHashes = new Set<string>()
+
+    pendingTransactions
+      .toSorted(({ time: newerTime }, { time: olderTime }) => newerTime - olderTime)
+      .forEach(({ hash, time, account, to, from, assetHash, token, ...transfer }) => {
+        if (
+          !accounts.some(AccountHelper.predicate(account)) ||
+          !isWithinInterval(createNewDateByTime(time), { start: dateFrom, end: dateTo })
+        )
+          return
+
+        const { blockchain } = account
+        const service = bsAggregator.blockchainServicesByName[blockchain]
+
+        let txTemplateUrl: string | undefined
+        let addressTemplateUrl: string | undefined
+        let contractTemplateUrl: string | undefined
+
+        if (hasExplorerService(service)) {
+          txTemplateUrl = service.explorerService.getTxTemplateUrl()
+          addressTemplateUrl = service.explorerService.getAddressTemplateUrl()
+          contractTemplateUrl = service.explorerService.getContractTemplateUrl()
+        }
+
+        const event: TFullTransactionAssetEvent = {
+          hash: assetHash,
+          eventType: 'token',
+          amount: transfer.amount,
+          methodName: 'transfer',
+          hashUrl: contractTemplateUrl?.replace('{hash}', assetHash),
+          tokenType: 'generic',
+          token,
+          from,
+          fromUrl: addressTemplateUrl?.replace('{address}', from),
+          fromAccount: transfer.fromAccount,
+          to,
+          toUrl: addressTemplateUrl?.replace('{address}', to),
+          toAccount: transfer.toAccount,
+        }
+
+        const wasAlreadyInsertedHash = insertedPendingTransferHashes.has(hash)
+
+        if (wasAlreadyInsertedHash) {
+          const transaction = groupedData.get(hash)!
+
+          transaction.events.push(event)
+
+          return
+        }
+
+        groupedData.set(hash, {
+          txId: hash,
+          txIdUrl: txTemplateUrl?.replace('{txId}', hash),
+          block: 0,
+          date: createNewDateByTime(time).toISOString(),
+          invocationCount: 0,
+          notificationCount: 0,
+          networkFeeAmount: '0',
+          systemFeeAmount: '0',
+          blockchain,
+          isPending: true,
+          events: [event],
+        })
+
+        insertedPendingTransferHashes.add(hash)
+      })
+
+    const items = Array.from(groupedData.values()).sort(({ date: newerDate }, { date: olderDate }) => {
+      if (new Date(newerDate) > new Date(olderDate)) return -1
+      if (new Date(newerDate) < new Date(olderDate)) return 1
+
+      return 0
```

Comment:
Use a const to storage the newerDate and olderDate, so you don't need to instance a `new Date` two times for the same date

```
const items = Array.from(groupedData.values()).sort((a, b) => {
     const newerDate = new Date(a.date)
      const olderDate = new Date(b.date)
      if (newerDate > olderDate) return -1
      if (newerDate < olderDate) return 1

      return 0
  })
```